service/storage_proxy.cc

```diff
@@ -5735,6 +5736,14 @@ storage_proxy::query_singular(lw_shared_ptr<query::read_command> cmd,
     exec.reserve(partition_ranges.size());
 
     schema_ptr schema = local_schema_registry().get(cmd->schema_version);
+    if (schema->is_view() && !schema->view_info()->base_info()) {
+        // If this is a reverse read, the reverse schema might have just been added
+        // to the schema registry, and the base info might not be set yet, so we set it here.
+        // Reverse reads are not supported for partition range queries,
+        // so query_partition_key_range() doesn't require this adjustment.
+        schema_ptr base_schema = local_db().find_schema(schema->view_info()->base_id());
+        schema->view_info()->set_base_info(schema->view_info()->make_base_dependent_view_info(*base_schema));
```

Comment:
Kudos for finding the root of this bug and this fix, but it's sad we (have to) have such nonsense in our code. I never understood why the "schema registry" even exists, or why reading from a view table needs to know the schema of the base table (it probably doesn't - a view is an independently-readable table!), and now we have more of this stuff inside storage_proxy.cc. But at this stage, I don't see another option.

Reply:
> I never understood why the "schema registry" even exists

It exists because we need a way to lookup schema_ptr by its version. There can be many versions in use at a time, we cannot take schema_ptr from table::_schema. Fetching non-current schema version is expensive (network RPC, across-shard deserialization), so we want to cache them. We also don't want to keep them forever. Hence registry.


> why reading from a view table needs to know the schema of the base table

It's used by updates at least, and we don't keep separate schema object just for reads.